frame/vamm/src/tests.rs

```diff
@@ -43,7 +41,33 @@ const MINIMUM_RESERVE: Balance = ZERO_RESERVE + 1;
 const MAXIMUM_RESERVE: Balance = Balance::MAX;
 
 #[allow(dead_code)]
-const RUN_CASES: u32 = 100;
+const RUN_CASES: u32 = 1000;
+
+prop_compose! {
+	fn min_max_reserve()(
+		base_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE,
+		quote_asset_reserves in MINIMUM_RESERVE..=MAXIMUM_RESERVE,
+		peg_multiplier in MINIMUM_RESERVE..=MAXIMUM_RESERVE
+	) -> (Balance, Balance, Balance) {
+		(base_asset_reserves, quote_asset_reserves, peg_multiplier)
+	}
+}
+
+prop_compose! {
+	fn zero_reserve()(
+		zero_reserve in ZERO_RESERVE..=ZERO_RESERVE,
+	) -> Balance {
+		zero_reserve
+	}
+}
```

Comment:
Can't you remove the arguments and just return `Balance::zero()`? If not, maybe do
```suggestion
prop_compose! {
	fn zero_reserve()(zero_reserve in Just(Balance::zero())) -> Balance {
		zero_reserve
	}
}
```

Reply:
Using `zero_reserve in Just(Balance::zero())` is way better.
Didn't know one could write it in this way, thanks for the input! :smile: